### PR TITLE
FEAT: SQLite binding for Red and Red/System (initial version)

### PR DESCRIPTION
**Pull request by Oldes**
SQLite is an in-process library that implements a self-contained, serverless, zero-configuration, transactional SQL database engine.

### CODE REVIEW

File: Library/SQLite/SQLite3.red [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+	Title:   "Red SQLite3 binding"
+	Author:  "Oldes"
+	File: 	 %SQLite3.red
+	Rights:  "Copyright (C) 2017 David 'Oldes' Oliva. All rights reserved."
+	License: "BSD-3 - https:;//github.com/red/red/blob/master/BSD-3-License.txt"
+]
+#system [
+	#include %SQLite3.reds
+
+	#define TRACE(value) [
```
Comment (dockimbel) [inline Library/SQLite/SQLite3.red:11]: It's a very bad idea to create R/S macro name that could easily collide with others code, or even the Red runtime library in the future.

**Review by dockimbel — APPROVED** [review]

File: Library/SQLite/SQLite3.red (hunk 1 above)
Comment (Oldes) [reply Library/SQLite/SQLite3.red:11]: Maybe I'm setting it as a standard:) What other name I should use? The trick of this approach is, that I comment out the print line and all traces are removed from the compiled code.

Comment (Oldes) [reply Library/SQLite/SQLite3.red:11]: But as I think about it... I'm already using it in several projects, so I would have conflicts with myself.

Comment (qtxie) [reply Library/SQLite/SQLite3.red:11]: Better to add prefix or suffix, and `_`. Something like: `SQL_TRACE`.

**Pull request merged**